lib/client/rpc/balances.ts

```diff
@@ -81,100 +95,121 @@ export class FutarchyRPCBalancesClient implements FutarchyBalancesClient {
    * @returns
    */
   async fetchAllConditionalTokenWalletBalances(
-    dao: DaoWithTokens,
     ownerWallet: PublicKey,
-    proposalsWithVaults: ProposalWithVaults[]
+    baseToken: TokenProps,
+    quoteToken: TokenProps,
+    proposals: ProposalWithVaults[]
   ): Promise<TokenWithBalanceWithProposal[]> {
-    if (ownerWallet && dao.baseToken.publicKey && dao.quoteToken.publicKey) {
-      const tokensWithPDA = proposalsWithVaults
-        .map((p) => {
-          return [
+    const tokenBalances = await Promise.all(
+      proposals.map(async (proposalWithVaults) => {
+        if (
+          ownerWallet &&
+          proposalWithVaults.publicKey &&
+          quoteToken.publicKey
+        ) {
+          const tokensWithPDA = [
```

Comment:
I feel like that building this into an object with some logic around it might make it more readable. But not going to fight on this given the demand right now.